crowbar_framework/app/models/service_object.rb

```diff
@@ -1576,4 +1616,28 @@ def profile(name, &block)
       block.call
     end
   end
+
+  def equal_or_match(value, maybe_regexp)
+    operator = lambda do
+      op, check = maybe_regexp.split(" ", 2)
+      value.to_f.send(op.to_sym, check.to_f)
+    end
+    
+    case
+    when maybe_regexp.start_with?('/') && maybe_regexp.end_with?('/')
+      Regexp.new(maybe_regexp[1..-2]).match(value)
+    when maybe_regexp.start_with?('==')
```

Comment:
If there is not operator, there is not conversion AFAICT. Is using a simple string comparison. If is prefixed with one operator (like '==') there is conversion to_f.